credentialdigger/client.py

```diff
@@ -1378,3 +1381,64 @@ def update_similar_snippets(self,
                 self.update_discovery(d['id'], state)
                 n_updated_snippets += 1
         return n_updated_snippets
+
+    def assign_categories(self, discoveries):
+        """ Add category to each discovery
+
+        Parameters
+        ----------
+        discoveries: list
+            List of discoveries without assigned categories to them
+
+        """
+        rulesdict = self.get_rules()
+        for discovery in discoveries:
+            if discovery['rule_id']:
+                category = rulesdict[discovery['rule_id'] - 1]['category']
+                discovery['category'] = category
+            else:
+                discovery['category'] = '(rule deleted)'
+
+    def export_discoveries_csv(self, repo_url, states='all'):
+        """ Export discoveries as CSV
+
+        Parameters
+        ----------
+        repo_url: str
+            The repo containing the discoveries
+        sates: str | list
+            - str: if it equals 'all', then return all discoveries.
+                   return chosen state otherwise (i.e 'false_positive')
+            - list: return all the discoveries that have states contained
+                    in this list (i.e ['new', 'false_positive'])
+
+        Returns
+        str
+            Discoveries in CSV format.
```

Comment:
CSV format? you wrote str above. Is str the correct type?

Reply:
Yes, str is the correct return type. 
CSV (comma-separated values) is a string itself, it is a string that has a certain format. So in the end, I believe that it is safe to call CSV a string.